src/token_manager.hpp

```diff
@@ -0,0 +1,64 @@
+#include "token_manager.h"
+#include "grpc_client.h"
+
+namespace grpc_labview
+{
+    template <typename T>
+    TokenManager<T>::TokenManager() :
+        _pointerForToken(),
+        _mutex()
+    {
+    }
+
+    template <typename T>
+    template <typename TDerivedType>
+    std::shared_ptr<TDerivedType> TokenManager<T>::GetPtrForToken(TokenManager::token_type token, int32_t* status)
+    {
+        std::lock_guard<std::mutex> lock(_mutex);
+        auto ptr = _pointerForToken.find(token);
+        if (ptr == _pointerForToken.end())
```

Comment:
RFC: I don't know the right way to handle errors. There was a lot of "return -1" already in grpc_client.cc.

1. If there's a better error range to use, I'd be happy to do so. Maybe one of the [gRPC status codes](https://grpc.github.io/grpc/core/md_doc_statuscodes.html), possibly along with the `-(1000 + status)` that grpc_client already uses in some places.
2. Maybe we don't need a status here. Just return the default pointer like I already am and let the client code check for an invalid pointer. This is essentially what I've already got implemented since all of the calling code that I changed is passing `nullptr` for the `status` pointer.
3. I don't know if writing to std::cerr is a good thing here either. I don't like hardcoding non-localized strings in code.